src/main/java/blusunrize/immersiveengineering/common/crafting/serializers/MineralMixSerializer.java

```diff
@@ -83,9 +95,20 @@ public void toNetwork(FriendlyByteBuf buffer, MineralMix recipe)
 		PacketUtils.writeList(buffer, recipe.spoils, StackWithChance::write);
 		buffer.writeInt(recipe.weight);
 		buffer.writeFloat(recipe.failChance);
-		buffer.writeInt(recipe.dimensions.size());
-		for(ResourceKey<DimensionType> dimension : recipe.dimensions)
-			buffer.writeResourceLocation(dimension.location());
+		buffer.writeInt(recipe.biomeTagPredicates.size());
+		for(BiomeTagPredicate biomes : recipe.biomeTagPredicates)
+		{
+			buffer.writeInt(biomes.tags().size());
+			for(TagKey<Biome> tag : biomes.tags())
+				buffer.writeResourceLocation(tag.location());
+		}
 		PacketUtils.writeRegistryElement(buffer, BuiltInRegistries.BLOCK, recipe.background);
 	}
+
+	private static Codec<BiomeTagPredicate> makeBiomePredicateCodec()
+	{
+		return RecordCodecBuilder.create(inst -> inst.group(
+				NeoForgeExtraCodecs.setOf(TagKey.codec(Registries.BIOME)).fieldOf("biome_tags").forGetter(BiomeTagPredicate::tags)
+		).apply(inst, BiomeTagPredicate::new));
```

Comment:
```java
NeoForgeExtraCodecs.setOf(TagKey.codec(Registries.BIOME))
    .xmap(BiomeTagPredicate::tags, BiomeTagPredicate::new)
```
?

Reply:
I don't know how codecs work!!!
I just experimented and this one worked. But I'll try that =P